lib/netext/httpext/request.go

```diff
@@ -66,6 +67,24 @@ func (u URL) GetURL() *url.URL {
 	return u.u
 }
 
+// Return a sanitized URL, clean of e.g. user credentials
+func cleanURL(urlString string) string {
+	u, err := url.ParseRequestURI(urlString)
+	if err == nil && u.User != nil {
+		// it's a proper URL, mask credentials
+		old := fmt.Sprintf("%s@", u.User.String())
+		new := "****@"
+		_, passSet := u.User.Password()
+		if passSet {
+			new = fmt.Sprintf("****:%s", new)
+		}
+		clean := strings.Replace(u.String(), old, new, 1)
+		// Replace back any literal `${}` placeholders escaped by `u.String()`
+		return strings.Replace(clean, `$%7B%7D`, `${}`, -1)
```

Comment:
This seems like a bit of a nasty hack :smile: In general, I slightly dislike that we're having to parse the previously parsed URL here and do a bunch of shifty string operations to sanitize it. This shouldn't noticeably affect performance, even when we do it twice for each URL in the script, bit it just seems like a waste, especially given the fact that most URLs probably won't have credentials... 

Why are you reluctant to mess with the `NewURL()` constructor https://github.com/loadimpact/k6/blob/d385166a821a8cd98a4ab2c158b3982d87ee61a5/lib/netext/httpext/request.go#L59-L62
It seems like it would be relatively easy to just add another `SanitizedURL` property or something like that, which can just be the same string as the URL if there's no authentication, and sanitized from the already parsed URL if there is. In my head that won't be more difficult to trace than the current implementation, so I don't think it's a case of premature optimization, but I might be wrong :smile: @MStoykov, thoughts?

Reply:
I am of the opinion that given the amount of strings that the above  code will generate I am pretty sure it will have performance impact if we do it for ALL urls. IMHO we should at least check if we need to do anything at all before we even parse it back from string to URL ... given that we already have it as a URL. 
Also given that .. I am pretty sure it would be better if we just copy the URL , change the user and password and string it back ... 
And ... isn't Name supposed to stay if it's set by the user and if it's not it's the same as the URL so we don't need to do the work twice ? 
As a whole I think that maybe taking a step back and working directly with the URLs will be better than what you are currently trying to do

Reply:
> Why are you reluctant to mess with the NewURL() constructor

I guess I was trying to introduce the least amount of changes while I'm getting familiar, but you're both right that it would be more efficient to create a clean URL in the constructor once and avoid parsing it every time the tags are set.

I'll refactor this again, thanks for the feedback.

Reply:
@na-- @MStoykov Take a look at the latest commit 6ab3603b. I suppose it's cleaner and more efficient, but I don't like that `httpext.URL` now wraps 4 representations of potentially the same URL. It's a bit messy to me, but I'm not familiar enough with the codebase to make a concrete suggestion for fixing it.